message_ix_models/model/water/build.py

```diff
@@ -103,6 +103,159 @@ def cat_tec_cooling(context: Context) -> tuple[pd.DataFrame, list[str]]:
     return cat_tec, regions_df
 
 
+def share_map_cool(
+    share_keys: list[str],
+    type_tec_keys: list[str],
+    regions_df: list[str],
+    commodity_mapping: dict[str, str] = None,
```

Comment:
```suggestion
    commodity_mapping: Optional[dict[str, str]] = None,
```

This would be the correct type hint if the default is `None`. Otherwise, mypy would complain because `None` is not compatible with `dict[str, str]`.